Hunted Mobile/Hunted Mobile/Hunted Mobile/ViewModel/MapViewModel.cs

```diff
@@ -358,16 +358,19 @@ public class MapViewModel : BaseViewModel {
             Location playingUserLocation = mapModel.PlayingUser.Location;
             var newPlayer = new List<Player>();
 
-            foreach(Player player in data.Players) {
-                var gadgets = mapModel.Players.Where(p => p.Id == player.Id).FirstOrDefault()?.Gadgets;
-                if(gadgets != null) {
-                    player.Gadgets = gadgets;
-                }
+            foreach(PlayerBuilder builder in data.PlayerBuilders) {
+                var player = builder.ToPlayer();
+                if(player != null) {
+                    var gadgets = mapModel.Players.Where(p => p.Id == player.Id).FirstOrDefault()?.Gadgets;
+                    if(gadgets != null) {
+                        player.Gadgets = gadgets;
+                    }
```

Comment:
Of `gadgets != null` kan je ook afvangen in het model. Aangezien je `player.Gadgets` doet, kan je in de setters van `Gadgets` een if-statment plaatsen om te kijken of het `null` is.

Reply:
Ik weet zo niet waarom ik dat niet gedaan had dus wellicht dat dat een verbetering zou zijn.